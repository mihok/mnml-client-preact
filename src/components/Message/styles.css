/* Message Component */


/* Messenger Theme */

.Message__client--messenger {
  align-self: flex-end;
  background: rgb(10, 107, 239);
  border-radius: 8px 8px 8px 0;
  color: white;
  display: inline-block;
  font-size: 12px;
  letter-spacing: 1px;
  margin: 0;
  max-width: 95%;
  min-width: 160px;
  padding: 5px 0 5px 10px;
  list-style-type: none;
}


/* Container box for Avatar-op + Operator__Messenger */
.Message__operatorWrapper--messenger {
  align-items: flex-start;
  display: flex;
  justify-content: flex-end;
  margin: 0.5rem 0;
}

.Message__operator--messenger {
  align-self: flex-start;
  background: #e1e1e1; 
  border-radius: 8px 0px 8px 8px;
  color: #333;
  display: inline-block;
  font-size: 12px;
  letter-spacing: 1px;
  list-style: none;
  max-width: 70%;
  padding: 0.5rem;
  text-align: left;
}

.Message__avatar--messenger {
  width: 40px;
  height: 40px;
  padding-left: 4px; 
}


/* Float Theme */

/* Top level container around an <li> message */
.Message__box--float{
    margin: 7px 0;
    padding-right: 10px; /* move msgs away from scrollbar */
}

.Message__avatar--float {
  width: 40px;
  height: 40px;
  border-radius: 20px;
}

.Message__operatorWrapper--float {
  align-items: flex-start;
  display: flex;
  justify-content: space-between;
}

.Message__operator--float {
  margin: 0;
  list-style: none;
  width: 160px;
  float: left;
  text-align: left;
  background: #ffffff;
  color: #000000;
  margin-top: 4px;
  margin-left: 4px;
  box-shadow: #dddddd 1px 1px 8px 0;
  border-radius: 10px 0px 10px 0px;
  padding: 16px;
}

.Message__client--float {
  margin: 0;
  padding: 16px;
  list-style: none;
  width: 160px;
  margin-top: 4px;
  margin-bottom: 4px;
  box-shadow: #dddddd 1px 1px 8px 0;
  border-radius: 10px 10px 10px 0px;
  background: #0a6bef;
  float: left;
  text-align: left;
  color: #ffffff;
  }

