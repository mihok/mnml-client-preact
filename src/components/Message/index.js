import { h } from 'preact';
import PropTypes from 'prop-types';

import themer from '../ThemeProvider/themeHOC';

import './styles.css';

const Message = props => {
  // our message's content
  const content = props.content.map((msg, i) => <li key={i}>{msg}</li>);

  // our default message is a client message
  let message = (
    <ul className={`Message__client__${props.theme}`}>
      {content}
    </ul>
  );

  // if the iterated message is an operator; override `message`
  if (props.type === 'operator') {
    message = (
      <div className={`Message__operator-box__${props.theme}`}>
        <ul className={`Message__operator__${props.theme}`}>
          {content}
        </ul>

        <img
          alt="Operator"
          className={`Message__Avatar-op__${props.theme}`}
          src="http://placehold.it/40x40/"
        />
      </div>
    );
  }

  // TODO: Messages sent within a period of time should be aggregated.

  // incoming props, mesage.content is an array.
  return (
    <li className={`Message__box__${props.theme}`}>
      {message}
    </li>
  );
};

Message.propTypes = {
  theme: PropTypes.string,
  type: PropTypes.string,
  content: PropTypes.arrayOf(PropTypes.string),
};

export default themer(Message);
