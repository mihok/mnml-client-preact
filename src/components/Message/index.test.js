/* const render = require('preact-render-to-string');
 * const App = require('../thing');*/
import render from 'preact-render-to-string';
import { h } from 'preact';
import Message from '.';

describe('App component', () => {
  it('should render ', () => {
    const props = {
      content: [], // message content
    };

    const tree = render(<Message content={props.content} />);
    expect(tree).toMatchSnapshot();
  });
});
