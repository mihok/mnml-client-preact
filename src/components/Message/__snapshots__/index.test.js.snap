// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render  1`] = `"<li class=\\"Message__box__undefined\\"><ul class=\\"Message__client__undefined\\"></ul></li>"`;
